external/tools/modify_imgui_for_gearsvk.ps1

```diff
@@ -0,0 +1,292 @@
+	if ($line -match "^static void ImGui_ImplVulkan_SetupRenderState\(") {
+		$inSetupRenderState = 1
+		Write-Host "  - Modifying ImGui_ImplVulkan_SetupRenderState"
+	}
+
+	# Detect if in ImGui_ImplVulkan_RenderDrawData
+	if ($line -match "^void ImGui_ImplVulkan_RenderDrawData\(") {
+		$inRenderDrawData = 1
+		Write-Host "  - Modifying ImGui_ImplVulkan_RenderDrawData"
+	}
+
+	# Detect if in ImGui_ImplVulkan_CreateFontsTexture
+	if ($line -match "^bool ImGui_ImplVulkan_CreateFontsTexture\(") {
+		$inCreateFontsTexture = 1
+		Write-Host "  - Modifying ImGui_ImplVulkan_CreateFontsTexture"
+	}
+
+	# Detect if the function we have modified is left by relying on consistent code
+	# formatting, i.e. the closing '}' is at the beginning of the line
+	if (($inRenderDrawData -Or $inSetupRenderState -Or $inCreateFontsTexture) -And $line -match "^}") {
+		if ($inSetupRenderState) {
+			Write-Host "   -> modified" -ForegroundColor Green
+			$inSetupRenderState = 0
+		}
+		if ($inRenderDrawData) {
+			Write-Host "   -> modified" -ForegroundColor Green
+			$inRenderDrawData = 0
+		}
+		if ($inCreateFontsTexture) {
+			Write-Host "   -> modified" -ForegroundColor Green
+			$inCreateFontsTexture = 0
+		}
+	}
+
+	# Remove font descriptor set binding in ImGui_ImplVulkan_SetupRenderState
+	if ($inSetupRenderState) {
+		# Remove unecessary descriptor binding
+		if (($line -match "VkDescriptorSet desc_set\[1\] = \{ g_DescriptorSet \};") -Or
+			($line -match "vkCmdBindDescriptorSets\(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL\);")) {
+			continue
+		}
+		# Change comment, stating that descriptor sets are bound
+		if ($line -match "// Bind pipeline and descriptor sets:") {
+			# Get tabs/spaces up to the first '/' to nicely align the new comment
+			$matched = $line -match "^[^/]*"
+			$tabs = $Matches[0]
+			$fileContentModified += $tabs + "// Bind pipeline"
+			continue
+		}
+	}
+
+	# Add descriptor set binding in ImGui_ImplVulkan_RenderDrawData
+	if ($inRenderDrawData) {
+		if ($line -match "vkCmdSetScissor\(command_buffer, 0, 1, &scissor\);") {
+			# Add the scissor cmd line
+			$fileContentModified += $line
+			# Get tabs/spaces up to the 'v' of vkCmdSetScissor to nicely align the code
+			$matched = $line -match "^[^v]*"
+			$tabs = $Matches[0]
+			# Add descriptor binding code after the vkCmdSetScissor cmd
+			$fileContentModified += ''
+			$fileContentModified += $tabs + '// Bind texture descriptor set stored as ImTextureID'
+			$fileContentModified += $tabs + 'VkDescriptorSet desc_set[1] = { (VkDescriptorSet) pcmd->TextureId };'
+			$fileContentModified += $tabs + 'vkCmdBindDescriptorSets(command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, g_PipelineLayout, 0, 1, desc_set, 0, NULL);'
+			continue
+		}
+	}
+
+	# Change fonts texture id assignment to descriptor in ImGui_ImplVulkan_CreateFontsTexture
+	if ($inCreateFontsTexture) {
+		if ($line -match "io.Fonts->SetTexID\(\(ImTextureID\)\(intptr_t\)g_FontImage\);") {
```

Comment:
Maybe also here some safety-regex-measures could be applied. E.g. `\s*` means: "whitespaces, either zero or multiple".